Silicon/ElkhartlakePkg/Library/HeciInitLib/HeciPeiDxe.c

```diff
@@ -494,3 +495,135 @@ HeciArbSvnGetInfoMsg (
+  key hash, revocation status, validity.
+
+  @param[in, out] OemKeyStatus    Status of OEM keys
+
+  @retval EFI_SUCCESS             Command succeeded
+  @retval EFI_UNSUPPORTED         Current ME mode doesn't support this function
+  @retval EFI_DEVICE_ERROR        HECI Device error, command aborts abnormally
+**/
+EFI_STATUS
+HeciGetOemKeyStatus(
+  IN OUT  OEM_KEY_STATUS *OemKeyStatus
+  )
+{
+  EFI_STATUS               Status;
+  UINT32                   Length;
+  UINT32                   RecvLength;
+  OEM_KEY_STATUS_BUFFER    OemKeyStatusBuff;
+  UINT32                   MeMode;
+
+  Status = HeciGetMeMode (&MeMode);
+  if (EFI_ERROR (Status) || (MeMode != ME_MODE_NORMAL)) {
+    return EFI_UNSUPPORTED;
+  }
+
+  ZeroMem (&OemKeyStatusBuff, sizeof (OemKeyStatusBuff));
+  OemKeyStatusBuff.Request.MkhiHeader.Data           = 0;
+  OemKeyStatusBuff.Request.MkhiHeader.Fields.GroupId = MKHI_MCA_GROUP_ID;
+  OemKeyStatusBuff.Request.MkhiHeader.Fields.Command = MCA_GET_OEM_KEY_STATUS_CMD;
+  Length                                         = sizeof (OEM_KEY_STATUS_REQ);
+  RecvLength                                     = sizeof (OEM_KEY_STATUS_ACK);
+
+  Status = HeciSendwAck (
+                   HECI1_DEVICE,
+                   (UINT32 *) &OemKeyStatusBuff,
+                   Length,
+                   &RecvLength,
+                   BIOS_FIXED_HOST_ADDR,
+                   HECI_MCHI_MESSAGE_ADDR
+                   );
+
+  if (!EFI_ERROR (Status) &&
+      (OemKeyStatusBuff.Response.MkhiHeader.Fields.Command == MCA_GET_OEM_KEY_STATUS_CMD) &&
+      (OemKeyStatusBuff.Response.MkhiHeader.Fields.IsResponse == 1) &&
+      (OemKeyStatusBuff.Response.MkhiHeader.Fields.Result == 0)) {
+    CopyMem (OemKeyStatus, &OemKeyStatusBuff.Response.OemKeyStatus, sizeof(OEM_KEY_STATUS));
+  } else {
+    DEBUG ((DEBUG_ERROR, "HeciGetOemKeyStatus Result: %08x \n", OemKeyStatusBuff.Response.MkhiHeader.Fields.Result));
+    if (!EFI_ERROR (Status)) {
+      Status = EFI_DEVICE_ERROR;
+    }
+  }
+
+  return Status;
+}
+
+
+/**
+  This command is used to indicate to the FW to revoke a OEM key.
+  It should be executed when MBP data indicates that a key is pending for
+  revocation.
+
+  @retval EFI_SUCCESS             Command succeeded
+  @retval EFI_UNSUPPORTED         Current ME mode doesn't support this function
+  @retval EFI_DEVICE_ERROR        HECI Device error, command aborts abnormally
+**/
+EFI_STATUS
+HeciRevokeOemKey (
+  VOID
+  )
+{
+  EFI_STATUS               Status;
+  UINT32                   Length;
+  UINT32                   RecvLength;
+  OEM_KEY_REVOKE_BUFFER    OemKeyRevoke;
+  UINT32                   MeMode;
+  ME_BIOS_PAYLOAD          *MbpDataHob;
+  UINT8                    *DataPtr;
+  UINT32                   MbpDataHobLen;
+  LOADER_GLOBAL_DATA       *LdrGlobal;
+
+  //
+  // Get Mbp Data HOB
+  //
+  LdrGlobal = (LOADER_GLOBAL_DATA *) GetLoaderGlobalDataPointer ();

```

Comment:
Would it be better to move the MBP and Hob logic out of this API?
Let the caller figure out if HeciRevokeApi need to be invoked or not.

Anyway, CSE will perform revocation ONLY if it was instructed to do so via 'revocation extension' in OEM Key manifest.

Reply:
Sachin, Can I take this as improvement later? There were other API's in HECI lib which uses MBP data which are consumed by PSD. I am planning to add MbpData as separate platform lib and required components can consume it.

Reply:
Sachin, I have removed MbpData check here. As it would forced updated from user, check is not required.